### PR TITLE
GH1994: Update MSBuildVerbosityExtensions.cs

### PR DESCRIPTION
**Pull request by jozefizso**
- [x] Add `GetMSBuildVerbosity()` to convert verbosity from string to enum
- [x] Add tests for `GetMSBuildVerbosity()`

### CODE REVIEW

**Comment by devlead** [comment]
@jozefizso thanks for your contribution, please ping us when you think the PR is ready for review and we'll take a look at it.

Also please raise an issue to go with this PR according to our [Contribution Guidelines](https://github.com/cake-build/cake/blob/develop/CONTRIBUTING.md).

File: src/Cake.Common/Tools/MSBuild/MSBuildVerbosityExtensions.cs [hunk 1]
```diff
@@ -35,5 +35,29 @@ public static string GetMSBuildVerbosityName(this Verbosity verbosity)
...
+                case "minimal":
+                    return Verbosity.Minimal;
+                case "normal":
+                    return Verbosity.Normal;
+                case "detailed":
+                    return Verbosity.Verbose;
+                case "diagnostic":
+                    return Verbosity.Diagnostic;
+                default:
+                    throw new CakeException("Encountered unknown MSBuild build log verbosity.");
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/MSBuild/MSBuildVerbosityExtensions.cs]: Perhaps we could output known values here as well.

`Encountered unknown MSBuild build log verbosity 'foo'. Valid values are 'quiet', 'minimal', 'normal', 'detailed' and 'diagnostic'.`

File: src/Cake.Common/Tools/MSBuild/MSBuildVerbosityExtensions.cs [hunk 2]
```diff
@@ -35,5 +35,29 @@ public static string GetMSBuildVerbosityName(this Verbosity verbosity)
                     throw new CakeException("Encountered unknown MSBuild build log verbosity.");
             }
         }
+
+        /// <summary>
+        /// Gets the MSBuild <see cref="Verbosity"/> from string value.
+        /// </summary>
+        /// <param name="verbosity">The verbosity string value.</param>
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/MSBuild/MSBuildVerbosityExtensions.cs]: Can we add a `Remarks` section here with the valid verbosity values, so they appear in the documentation.

Comment (jozefizso) [reply src/Cake.Common/Tools/MSBuild/MSBuildVerbosityExtensions.cs]: I added description to remarks and exception message.

**Comment by jozefizso** [comment]
@devlead I think it is ready now.

**Comment by devlead** [comment]
@jozefizso your changes have been merged, thanks for your contribution 👍

**Pull request merged**